tests/SourcemapToolkit.SourcemapParser.UnitTests/SourceMapUnitTests.cs

```diff
@@ -118,5 +119,230 @@ public void GetMappingEntryForGeneratedSourcePosition_NoExactMatchHasSimilarOnDi
+			SourcePosition original2 = generateSourcePosition(2);
+			MappingEntry mapping = getSimpleEntry(generated2, original2, "sourceOne.js");
+
+			SourceMap map = new SourceMap
+			{
+				File = "generated.js",
+				Sources = new List<string> { "sourceOne.js" },
+				ParsedMappings = new List<MappingEntry> { mapping }
+			};
+
+			// Act
+			SourceMap combinedMap = map.ApplySourceMap(null);
+
+			// Assert (decorated expected exception)
+		}
+
+		[TestMethod]
+		public void ApplyMap_NoMatchingSources_ReturnsSameMap()
+		{
+			// Arrange
+			SourcePosition generated1 = generateSourcePosition(2);
+			SourcePosition original1 = generateSourcePosition(1);
+			MappingEntry childMapping = getSimpleEntry(generated1, original1, "someOtherSource.js");
+
+			SourceMap childMap = new SourceMap
+			{
+				File = "notSourceOne.js",
+				Sources = new List<string> { "someOtherSource.js" },
+				ParsedMappings = new List<MappingEntry> { childMapping }
+			};
+
+			SourcePosition generated2 = generateSourcePosition(3);
+			SourcePosition original2 = generateSourcePosition(2);
+			MappingEntry parentMapping = getSimpleEntry(generated2, original2, "sourceOne.js");
+
+			SourceMap parentMap = new SourceMap
+			{
+				File = "generated.js",
+				Sources = new List<string> { "sourceOne.js" },
+				ParsedMappings = new List<MappingEntry> { parentMapping }
+			};
+
+			// Act
+			SourceMap combinedMap = parentMap.ApplySourceMap(childMap);
+
+			// Assert
+			Assert.IsNotNull(combinedMap);
+			MappingEntry firstMapping = combinedMap.ParsedMappings[0];
+			Assert.IsTrue(firstMapping.IsValueEqual(parentMapping));
+		}
+
+		[TestMethod]
+		public void ApplyMap_MatchingSources_ReturnsCorrectMap()
+		{
+			// Arrange
+			SourcePosition generated1 = generateSourcePosition(2);
+			SourcePosition original1 = generateSourcePosition(1);
+			MappingEntry childMapping = getSimpleEntry(generated1, original1, "sourceTwo.js");
+
+			SourceMap childMap = new SourceMap
+			{
+				File = "sourceOne.js",
+				Sources = new List<string> { "sourceTwo.js" },
+				ParsedMappings = new List<MappingEntry> { childMapping }
+			};
+
+			SourcePosition generated2 = generateSourcePosition(3);
+			SourcePosition original2 = generateSourcePosition(2);
+			MappingEntry parentMapping = getSimpleEntry(generated2, original2, "sourceOne.js");
+
+			SourceMap parentMap = new SourceMap
+			{
+				File = "generated.js",
+				Sources = new List<string> { "sourceOne.js" },
+				ParsedMappings = new List<MappingEntry> { parentMapping }
+			};
+
+			// Act
+			SourceMap combinedMap = parentMap.ApplySourceMap(childMap);
+
+			// Assert
+			Assert.IsNotNull(combinedMap);
+			Assert.AreEqual(1, combinedMap.ParsedMappings.Count);
+			Assert.AreEqual(1, combinedMap.Sources.Count);
+			MappingEntry rootMapping = combinedMap.GetMappingEntryForGeneratedSourcePosition(generated2);
+			Assert.AreEqual(0, rootMapping.OriginalSourcePosition.CompareTo(childMapping.OriginalSourcePosition));
```

Comment:
so there's a mapping that points (3:5) to (2:4). There's another mapping in the submap from (2:4) to (1:3). What this is asserting is after applying the submap, (3:5) should now point to (1:3). Or did you want explicit line and column number checks?